src/wazuh_modules/vulnerability_scanner/src/scanOrchestrator/scanOrchestrator.hpp

```diff
@@ -257,46 +257,69 @@ class TScanOrchestrator final : public TOSPrimitives
         auto context = std::make_shared<TScanContext>(data);
         const auto type = context->getType();
 
+        // Postpone the event if:
+        // - Wasn't previously postponed
+        // - There are delayed events for that agent
+        // - The event is not a cleanup or re-scan all agents event
+        // Cleanup and re-scan events purge the delayed queue, so they have priority to be processed
         if (!isDelayed && type != ScannerType::CleanupAllAgentData && type != ScannerType::ReScanAllAgents &&
-            m_eventDelayedDispatcher->size(context->agentId()))
+            type != ScannerType::CleanupSingleAgentData && m_eventDelayedDispatcher->size(context->agentId()))
         {
+            logDebug2(WM_VULNSCAN_LOGTAG, "Postponing event (%d) for agent %s", type, context->agentId().data());
             m_eventDelayedDispatcher->push(context->agentId(), rawData);
         }
         else
         {
             switch (type)
             {
-                case ScannerType::HotfixInsert: m_hotfixInsertOrchestration->handleRequest(std::move(context)); break;
-                case ScannerType::PackageInsert: m_packageInsertOrchestration->handleRequest(std::move(context)); break;
-                case ScannerType::PackageDelete: m_packageDeleteOrchestration->handleRequest(std::move(context)); break;
-                case ScannerType::Os: m_osOrchestration->handleRequest(std::move(context)); break;
+                case ScannerType::HotfixInsert:
+                    logDebug2(WM_VULNSCAN_LOGTAG, "Hotfix inserted");
```

Comment:
I think we could improve the logging, put post and pre-processing, indicate if it is a delayed execution or not... these are examples.